tests/unit/externalRequestsTest.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const sinon = require('sinon')
+const tap = require('tap')
+
+const externalRequests = require('../../src/middleware/externalRequests')
+
+tap.test('External Requests Middleware', {autoend: true}, t => {
+  t.test('prepareLookupRequests', {autoend: true}, t => {
+    let performRequestsStub
+    t.beforeEach(done => {
+      performRequestsStub = sinon.stub(externalRequests, 'performRequests')
+      done()
+    })
+
+    t.afterEach(done => {
+      performRequestsStub.restore()
+      done()
+    })
+
+    t.test('should throw an error if any promise in the array fails', t => {
+      t.plan(1)
+      performRequestsStub
+        .onFirstCall()
+        .returns([Promise.resolve('Success'), Promise.reject('Fail')])
+
+      const ctx = {
+        state: {
+          uuid: 'randomUidForRequest',
+          metaData: {
+            name: 'Testing endpoint',
+            requests: {
+              lookup: [{}, {}]
```

Comment:
minor: lets maybe add a comment in each empty object that defines the stubbed outcome of this request that makes it a bit more clear why the object is empty
e,g
```
lookup: [
  {
    // first lookup request config - responds with success
  }, 
  {
    // second lookup request config - responds with fail
  }
]
```

Reply:
Done in https://github.com/jembi/openhim-mediator-mapping/pull/37/commits/1271b1e78f3643fd268cd6336803c17e7d1adcd4